feat: add clientConfig to from verb

sitaram-kalluri opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines in CONTRIBUTING.md

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**
- Send at_client version to server

**- How I did it**
- Pass the client version via the clientConfig in AtLookupImpl in the RemoteSecondary.

**- How to verify it**

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

sitaram-kalluri committed:
feat: add clientConfig to from verb

sitaram-kalluri committed:
build: updated CHANGELOG.md and version in pubspec.yaml

sitaram-kalluri committed:
Merge branch 'trunk' into add_clientConfig

gkc commented on at_client/lib/src/util/at_client_util.dart:
```diff
@@ -135,6 +134,13 @@ class AtClientUtil {
   }
+
+  static String getVersion() {
+    var applicationConfiguration = ApplicationConfiguration('pubspec.yaml');
```
This is dependent on what the "current working directory" is ...

Suggest you use this approach instead: https://stackoverflow.com/a/72486732

sitaram-kalluri replied:
Explored the above, the rootBundle is a flutter dependent package. Hence unable to use it in a Dart package.

sitaram-kalluri committed:
fix: add at_client_config

sitaram-kalluri committed:
Merge remote-tracking branch 'origin/add_clientConfig' into add_clientConfig

gkc commented on at_client/lib/src/preference/at_client_config.dart:
```diff
@@ -0,0 +1,13 @@
+  }
+
+  /// Represents the at_client version.
+  final String atClientVersion = '3.0.36';
```
Maintenance of this needs to be automated - maybe through using build_runner. At a minimum, you need to add a VERY VISIBLE comment to pubpsec.yaml stating that when the package version is increasing, then this constant needs to be changed accordingly.

sitaram-kalluri replied:
@gkc: For now, to get going, will add a comment in pubspec.yaml. And keep exploring the alternatives to automate the capturing the at_client version. Does this sound's good?

gkc replied:
👍 thanks Sita

sitaram-kalluri committed:
fix: add comment in pubspec.yaml and change e2e atSign from canary to trunk

sitaram-kalluri commented:
@gkc: Updated the pubspec.yaml with a comment. and changed the e2e test atSign from `ce2e4` to `ce2e2` because `ce2e4` runs on a prod version of at_secondary_server and hence the e2e tests are failing because the changes related to clientConfig are not yet released into Prod. I will a new PR having tests guarded with version checking to support cross version testing.

gkc reviewed (dismissed):
OK for now

sitaram-kalluri committed:
Merge branch 'trunk' into add_clientConfig
